# HG changeset patch
# User Mikhail Paulyshka <me@mixaill.tk>
# Date 1506252750 -10800
#      Sun Sep 24 14:32:30 2017 +0300
# Branch SDL2-WIN-SYMBOLS_LEACKAGE
# Node ID 46ec9baae30cd4e0c584de125cae4a3cce2864ad
# Parent  8df7a59b55283aa09889522369a2b32674c048de
win32: fix symbols leakage for static libraries

diff -r 8df7a59b5528 -r 46ec9baae30c include/begin_code.h
--- a/include/begin_code.h	Fri Sep 22 11:25:52 2017 -0700
+++ b/include/begin_code.h	Sun Sep 24 14:32:30 2017 +0300
@@ -58,8 +58,10 @@
 #   else
 #    define DECLSPEC    __declspec(dllimport)
 #   endif
+#  elif defined(_DLL)
+#   define DECLSPEC __declspec(dllexport)
 #  else
-#   define DECLSPEC __declspec(dllexport)
+#   define DECLSPEC
 #  endif
 # elif defined(__OS2__)
 #   ifdef BUILD_SDL
